autoparams/src/main/java/org/javaunit/autoparams/generator/ComplexObjectGenerator.java

```diff
@@ -11,22 +11,41 @@
 import java.util.Comparator;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Stack;
 import java.util.stream.Stream;
 import org.javaunit.autoparams.Builder;
 
 final class ComplexObjectGenerator implements ObjectGenerator {
+    private static final long MAX_RECURSION_COUNT = 3;
+    private final Stack<Type> objectGenerationStack = new Stack<>();
```

Comment:
I think it would be better to go into ObjectGenerationContext. But I didn't want to even consider the interface design, so I declared it inside the ComplexObjectGenerator.